dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -301,6 +309,19 @@ protected String createValidationMessages(U version) {
         return result.toString();
     }
 
+    protected String invalidFileNames(U version) {
```

Comment:
Minor, optional stuff:

1. Plural method name implies it returns a list/collection, but you return a string. I would either rename to something like `invalidFileNamesMessage` or leave the name and return a `List<String>` and let the caller convert to a string.
2. In general, not sure if so for hosted entries, filenames can contain spaces, so I would separate them with a comma-space rather than a space.

Even more optional, but I prefer a more functional way of writing this, untested:

```
return sourceFiles.stream()
  .map(sf -> sf.getPath())
  .filter(path -> path.endsWith("/"))
  .collect(Collectors.joining(", "));
  //.collect(Collectors.toList()); // If you wanted to return a list instead